apps/preview/next/pages/showcases/OrderSummary/OrderSummary.tsx

```diff
@@ -23,69 +54,140 @@ export default function OrderSummary() {
+  const removePromoCode = () => {
+    setPromoCode(0);
+    setInformationAlert(true);
+  };
 
   return (
-    <div className="md:shadow-lg md:rounded-md md:border md:border-neutral-100">
-      <div className="flex justify-between items-end bg-neutral-100 md:bg-transparent py-2 px-4 md:px-6 md:pt-6 md:pb-4">
-        <p className="typography-headline-4 font-bold md:typography-headline-3">Order Summary</p>
-        <p className="typography-text-base font-medium">(Items: {orderDetails.items})</p>
-      </div>
-      <div className="px-4 pb-4 mt-3 md:px-6 md:pb-6 md:mt-0">
-        <div className="flex justify-between typography-text-base pb-4">
-          <div className="flex flex-col grow pr-2">
-            <p>Items Subtotal</p>
-            <p className="typography-text-xs text-neutral-500">Original Price</p>
-            <p className="typography-text-xs text-secondary-700">Savings</p>
-            <p className="my-2">Delivery</p>
-            <p>Estimated Sales Tax</p>
+    <div>
+      <div className="md:shadow-lg md:rounded-md md:border md:border-neutral-100">
+        <div className="flex justify-between items-end bg-neutral-100 md:bg-transparent py-2 px-4 md:px-6 md:pt-6 md:pb-4">
+          <p className="typography-headline-4 font-bold md:typography-headline-3">Order Summary</p>
+          <p className="typography-text-base font-medium">(Items: {orderDetails.items})</p>
+        </div>
+        <div className="px-4 pb-4 mt-3 md:px-6 md:pb-6 md:mt-0">
+          <div className="flex justify-between typography-text-base pb-4">
+            <div className="flex flex-col grow pr-2">
+              <p>Items Subtotal</p>
+              <p className="typography-text-xs text-neutral-500">Original Price</p>
+              <p className="typography-text-xs text-secondary-700">Savings</p>
+              <p className="my-2">Delivery</p>
+              <p>Estimated Sales Tax</p>
+            </div>
+            <div className="flex flex-col text-right">
+              <p>{formatPrice(itemsSubtotal())}</p>
+              <p className="typography-text-xs text-neutral-500">{formatPrice(orderDetails.originalPrice)}</p>
+              <p className="typography-text-xs text-secondary-700">{formatPrice(orderDetails.savings)}</p>
+              <p className="my-2">{formatPrice(orderDetails.delivery)}</p>
+              <p>{formatPrice(orderDetails.tax)}</p>
+            </div>
+          </div>
+          {promoCode ? (
+            <div className="flex items-center mb-5 py-5 border-y border-neutral-200">
+              <p>PromoCode</p>
+              <SfButton size="sm" variant="tertiary" className="ml-auto mr-2" onClick={removePromoCode}>
+                Remove
+              </SfButton>
+              <p>{formatPrice(promoCode)}</p>
+            </div>
+          ) : (
+            <form
+              className="flex gap-x-2 py-4 border-y border-neutral-200 mb-4"
+              onSubmit={(event) => checkPromoCode(event)}
```

Comment:
No need to do it this way. You can pass function as is and it will be invoked with the event.
```suggestion
              onSubmit={checkPromoCode}
```

Reply:
https://github.com/vuestorefront/storefront-ui/pull/2627/commits/eb6f7745808cba73336afdd131c5e835dadbcee6